relational_operators/WorkOrderFactory.cpp

```diff
@@ -533,11 +530,13 @@ bool WorkOrderFactory::ProtoIsValid(const serialization::WorkOrder &proto,
         return false;
       }
 
+      const CatalogRelationSchema &build_relation = catalog_database.getRelationSchemaById(build_relation_id);
       const CatalogRelationSchema &probe_relation = catalog_database.getRelationSchemaById(probe_relation_id);
       for (int i = 0; i < proto.ExtensionSize(serialization::HashJoinWorkOrder::join_key_attributes); ++i) {
         const attribute_id attr_id =
             proto.GetExtension(serialization::HashJoinWorkOrder::join_key_attributes, i);
-        if (!probe_relation.hasAttributeWithId(attr_id)) {
+        if (!build_relation.hasAttributeWithId(attr_id) ||
```

Comment:
Note that I had removed the check for `build_relation`, as the `attr_id` in `probe_relation` may be different to that in `build_relation`, although both refer to the same `CatalogAttribute`.